### ISSUE TITLE
Memory leak on GPU 0

### ISSUE THREAD

**Issue by iwalkdaline** [NVIDIA/DIGITS#922]
I have a memory leak on GPU 0 where I'm consuming about 9 / 12 GB of memory on my Titan X.  The only process that is running on this GPU is digits-devserver.  I was wondering how much memory does the digits process consume on average.  This might be a leak that I caused but I'm not sure why the scheduler would allocate this amount of memory to a dataset task that I'm running on the CPU only.

**Comment by IsaacYangSLA** [comment]
@iwalkdaline , can you let us know the result of nvidia-smi when you do training on DIGITS?  That application tells how much memory each process uses.  Usually, the GPU is utilized by underlying framework, such as caffe.  If you are using bvlc caffe and it uses lots of memory, you might need to report that to BVLC/caffe.

**Comment by iwalkdaline** [comment]
I'm not sure I fully understand your reply.  I have done nvidia-smi which is how I discovered that GPU 0 was "using" 9 GB and the only process running on it is digits-server (PID 14690).  I have attached a screenshot for clarification.

![screen shot 2016-07-21 at 4 21 40 pm](https://cloud.githubusercontent.com/assets/6350235/17037507/4a0583e6-4f5f-11e6-9e98-1ce328d49613.png)

**Comment by IsaacYangSLA** [comment]
@iwalkdaline , we had an internal discussion about your issue, but couldn't pin point the root cause clearly without more information.  Will it be possible for you to provide us answers to the following questions?  Thanks.
1.	Can you help show the entire nvidia-smi result?  A few minutes of nvidia-smi -l 4 will be even better.
2.	If that during training or testing?
3.	Nv-caffe or BLVC Caffe?
4.     Was there a previous training / testing run on the GPU?

**Comment by iwalkdaline** [comment]
@IsaacYangSLA: I identified the source of the problem.  It seems that "classify many" has a memory leak.  When the task is run, it consumes about 3 Gb of memory on GPU 0 and never frees that memory.  Everytime it's run it seems to consume an additional 3 Gb of memory.  Not sure what happens when the limits of the GPU memory are reached in that case.  However, I did realize this would cause the scheduler to allocate jobs to GPU 0 and then the memory assertion/error would be thrown. This was because the scheduler didn't know 9 Gb of the memory was locked up.  This would then escalate to critical status only when a model is training on GPU 0 and then "classify many" was run.  The "classify many" task would collide with the running model and cause digits server to crash.  This bug seems fixed in the 4.0 release.  Thanks!
